src/main/java/org/mule/extension/file/internal/DirectoryListener.java

```diff
@@ -210,7 +214,7 @@
 
   @Override
   public void onStart(SourceCallback<InputStream, ListenerFileAttributes> sourceCallback) throws MuleException {
-    if (!muleContext.isPrimaryPollingInstance()) {
+    if (!clusterService.isPrimaryPollingInstance()) {
```

Comment:
should also consider this class's use of PrimaryNodeLifecycleNotificationListener